docs/pages/components/step-input.md

```diff
@@ -0,0 +1,445 @@
+                onclick="stepInputValue('step-4', 'up');" 
+                type="button">
+            </button>
+    </div>
+    
+</div>
+{% endcapture %}
+{% include display-component.html component=step-input-cozy %}
+
+
+## Compact Step Input
+Step Input can be used in compact mode, which is done to be used in desktop, or larger screen devices. 
+it can be achieved by adding `--compact` modifier to main element, button and input elements.
+
+{% capture step-input-compact %}
+<div class="fd-form-item fd-form-item--horizontal">
+    <div class="fd-step-input fd-step-input--compact">
+            <button aria-label="Step down" class="
+                fd-button
+                fd-button--compact
+                fd-button--transparent
+                fd-step-input__button
+                sap-icon--less"
+                onclick="stepInputValue('step-1', 'down');"
+                type="button">
+            </button>
+        <input class="
+                fd-input 
+                fd-input--compact 
+                fd-input--no-number-spinner 
+                fd-step-input__input
+        " id="step-1" type="number" value="0">
+            <button aria-label="Step up" class="
+                fd-button 
+                fd-button--compact
+                fd-button--transparent
+                fd-step-input__button
+                sap-icon--add"
+                onclick="stepInputValue('step-1', 'up');" 
+                type="button">
+            </button>
+    </div>
+    <span class="fd-form-label fd-form-label--unit-description">PC</span>
+</div>
+<br><br>
+<h3>RTL Support</h3>
+<div class="fd-form-item fd-form-item--horizontal" dir="rtl">
+    <div class="fd-step-input fd-step-input--compact">
+            <button aria-label="Step down" class="
+                fd-button
+                fd-button--compact
+                fd-button--transparent
+                fd-step-input__button
+                sap-icon--less"
+                onclick="stepInputValue('step-2', 'down');"
+                type="button">
+            </button>
+        <input class="
+                fd-input 
+                fd-input--compact 
+                fd-input--no-number-spinner 
+                fd-step-input__input
+        " id="step-2" type="number" value="0">
+            <button aria-label="Step up" class="
+                fd-button 
+                fd-button--compact
+                fd-button--transparent
+                fd-step-input__button
+                sap-icon--add"
+                onclick="stepInputValue('step-2', 'up');" 
+                type="button">
+            </button>
+    </div>
+    <span class="fd-form-label fd-form-label--unit-description">PC</span>
+</div>
+{% endcapture %}
+{% include display-component.html component=step-input-compact %}
+
+## Step Input with Different states
+Step Input Component also supports semantic states, same as for every form. They can be achieved by adding
```

Comment:
The Step Input component also supports semantic states as does every form. The semantic states can be customized by adding the `is-error` | `is-success` | `is-warning` | or `is-information`  into the `fd-step-input` element.